extensions/rocksdb-repos/database/RocksDbInstance.h

```diff
@@ -0,0 +1,71 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include "utils/OptionalUtils.h"
+#include "RocksDbUtils.h"
+#include "rocksdb/db.h"
+#include "logging/Logger.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace internal {
```

Comment:
Did you consider putting the extension, or the new classes, under a different namespace? e.g. `org::apache::nifi::minifi::rocksdbrepos`

Reply:
it is certainly doable, should we go with "rocksdbrepos"?

Reply:
I'm fine with either that, `extensions::rocksdbrepos`, `extensions::rocksdb` or anything similar.

Reply:
as I started refactoring, I am now leaning more towards leaving it as is, if you don't feel strongly about it

Reply:
ok, that's fine